internal/planner/languages/haskell/haskell_v1_planner.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package haskell
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"go.jetpack.io/devbox/internal/planner/plansdk"
+)
+
+var haskellPackageRegex = regexp.MustCompile(`^haskellPackages.[a-zA-Z0-9\-\_\.]+$`)
+
+// TODO: We should also support different GHC versions.
+// These can take the format of `haskell.compiler.ghc90` or similar
+var ghcRegex = regexp.MustCompile(`^ghc$`)
+
+var stackRegex = regexp.MustCompile(`^stack$`)
+var cabalRegex = regexp.MustCompile(`^cabal-install$`)
+
+type V2Planner struct {
+	userPackages []string
+}
+
+var _ plansdk.PlannerForPackages = (*V2Planner)(nil)
+
+func (p *V2Planner) Name() string {
+	return "haskell.v2.Planner"
+}
+
+func (p *V2Planner) IsRelevant(srcDir string) bool {
+	return false
+}
+
+func (p *V2Planner) IsRelevantForPackages(packages []string) bool {
+	p.userPackages = packages
```

Comment:
out of scope for this PR, but this is a strange API, where we save `packages` as a side-effect in a boolean IsRelevantForPackages function call....

Reply:
It's also weird that we do 2 passes of the regex, one to detect if the planner is relevant, and another to actually create the shell plan -- should we be storing the results of the `IsRelevantForPackages` and then reusing it in `GetShellPlan`?